Document how to locally run Athenian (#53)
athenianco/athenian-webapp: docs/CONTRIBUTING.md

For locally testing the authentication, you need to configure Auth0 and GitHub to work together. Follow these steps or refer to Auth0 and GitHub docs.

> You also need to run the [Athenian API](https://github.com/athenianco/athenian-api) and configure it in order to connect with your "Auth0 Management API"

Reviewer: ```suggestion
You also need to run the [Athenian API](https://github.com/athenianco/athenian-api) and configure it in order to connect to your "Auth0 Management API"
```

- "Allowed Web Origins": `http://localhost:3000`
- "Allowed Logout URLs": `http://localhost:3000`

#### 2. Create a Machine-to-Machine Applications in Auth0 ([docs](https://auth0.com/docs/dashboard/guides/applications/register-app-m2m))

Create an Application with "Application Type" set to "Machine to Machine"
- When asked for an API you want to authorize, choose "Auth0 Management API"
- "Scopes": `read:users`
- You don't need to configure any other value

#### 3. Connect your Auth0 app to a GitHub OAuth App ([docs](https://auth0.com/docs/connections/social/github))

1. Register a new OAuth App in GitHub,  with these settings:
    - "Homepage URL": `http://localhost:3000`
    - "Authorization callback URL": `https://<auth0 domain>/login/callback` (e.g., `your-app.eu.auth0.com`)
2. Enable the connection with GitHub from your Auth0 Dashboard providing the `Client ID` and `Client Secret` of the GitHub OAuth App from the previous step, and ask for these permissions:
    - `read:user`

#### 4. Configure Athenian WebApp

Update the [`config.js`](https://github.com/athenianco/athenian-webapp/blob/master/public/config.js) file:

> - `auth.domain`, `auth.clientId`, `auth.redirectUri` and `auth.logoutRedirectUri` with the corresponding values of your Auth0 Single Page Application created by `1`.

Author: ```suggestion
- `auth.domain`, `auth.clientId`, `auth.redirectUri` and `auth.logoutRedirectUri` with the corresponding values of your Auth0 Single Page Application created by [#1](#1-create-a-single-page-applications-in-auth0-docs).
```

- `auth.audience` will be the one for your "Auth0 Management API" (e.g. https://your-tenant.eu.auth0.com/api/v2/).
- `api.basePath` must be the url of your [Athenian API](https://github.com/athenianco/athenian-api)

#### 5. Configure Athenian API
Configure your [Athenian API](https://github.com/athenianco/athenian-api) (using the `.env` file if you're runing it locally):

> - `AUTH0_DOMAIN`, `AUTH0_CLIENT_ID` and `AUTH0_CLIENT_SECRET` with the corresponding values of your Auth0 Machine to Machin Application created by `2`.

Author: ```suggestion
- `AUTH0_DOMAIN`, `AUTH0_CLIENT_ID` and `AUTH0_CLIENT_SECRET` with the corresponding values of your Auth0 Machine to Machine Application created in [#2](#2-create-a-machine-to-machine-applications-in-auth0-docs).
```
